### PR TITLE
[Gitlab-CI] Automatic deployment of the standard library documentation to GH-pages

### PR DESCRIPTION
**Pull request by vbgl**
There are two objectives:
  - deploying the documentation of the standard library on *protected* branches
  - ease the merging of #9475 to branch 8.9

### CODE REVIEW

**Review by ejgallego — APPROVED** [review]

**Comment by ejgallego** [comment]
I couldn't see the pipelines in your fork @vbgl , merging then.

**Comment by Zimmi48** [comment]
Something annoying is that every time the tests fail for some reason (e.g. overlay not yet merged), the deploying stage is not run. There is also the risk that if we relaunch a failed build, a more recent version of the documentation will be overwritten by an older one...

**Comment by vbgl** [comment]
It is annoying, I agree. Thanks for raising the issue.

Do you think it’s a big deal?

**Comment by Zimmi48** [comment]
No, I don't think so. I don't see any better solution than waiting for GitLab CI to implement DAG-style pipelines (see https://gitlab.com/gitlab-org/gitlab-ce/issues/47063). As for the problem of overwriting, let's just not restart failed jobs when a newer job on the same branch has run to completion. It would be good though to have a more precise version number show on the documentation page (in the style of #483).

**Comment by vbgl** [comment]
What about allowing third-party developments to fail when running CI on `master` (or all protected branches)?

We would get the same information as before, but would allow the deployment stage to run.

**Comment by Zimmi48** [comment]
That could be an option but I'm unsure we would actually be informed of the failures in the same way.
Ideally, I'd rather make sure that we reduce the frequency of these failures, e.g. by seriously considering proceeding with something like #6724.

**Comment by SkySkimmer** [comment]
>What about allowing third-party developments to fail when running CI on master (or all protected branches)?

I don't think we can do that, allow_failure is unconditional.

**Comment by ejgallego** [comment]
Failures should not be allowed, that's the real problem and we should not try to workaround it; mtac people is using bors , other plugins should do the same or give developers commit access.

**Pull request merged**
